Fix links and update footnotes SDKv5 - Partially closes #1323 (#1328)
LiskHQ/lisk-docs: modules/ROOT/pages/introduction/blockchain-applications.adoc

In short, the on-chain logic:

* introduces state mutations through blocks footnote:block_footnote[For more information about blocks, check out the xref:{url_protocol_blocks}[] page of the Lisk protocol.] or transactions footnote:tx_footnote[For more information about transactions, check the xref:{url_protocol_transactions}[] page of the Lisk protocol.].
* is part of the blockchain protocol.

> * is verifiable by anyone by synchronizing with the blockchain.

Reviewer: ```suggestion
* is verifiable by anyone just by synchronizing with the blockchain.
```

For example, a new module could be added to the application, which accepts smart contracts.
This way it is possible to have the best of both worlds combined.

Additionally, smart contracts can reuse an already existing blockchain, which saves time when launching the application, as it is not necessary to take care of setting up an independent blockchain network, finding delegates footnote:delegate_footnote[For more information about delegates, check the xref:{url_protocol_consensus}[Consensus algorithm] page of the Lisk protocol.], etc.

Blockchain applications on the contrary rely on their own blockchain, and therefore also need to take care of maintaining their own network.

> In the beginning, this will make the launch of the application a little more complex, but having an independent network comes along with numerous benefits, which are covered in the following paragraphs.

Reviewer: ```suggestion
In the beginning, this will make the launch of the application slightly more complex, however, having an independent network comes with numerous benefits which are covered in the following paragraphs.
```
